pkg/proc/disasm_arm64.go

```diff
@@ -0,0 +1,247 @@
+	if inst.Inst == nil {
+		return "?"
+	}
+	var text string
+
+	switch flavour {
+	case GNUFlavour:
+		text = arm64asm.GNUSyntax(arm64asm.Inst(*inst.Inst))
+	case GoFlavour:
+		text = arm64asm.GoSyntax(arm64asm.Inst(*inst.Inst), inst.Loc.PC, bi.symLookup, readerAt)
+	case IntelFlavour:
+		fallthrough
+	default:
+		text = arm64asm.GNUSyntax(arm64asm.Inst(*inst.Inst))
+	}
+
+	return text
+}
+
+// IsCall returns true if the instruction is a CALL or LCALL instruction.
+func (inst *AsmInstruction) IsCall() bool {
+	if inst.Inst == nil {
+		return false
+	}
+	return inst.Inst.Op == arm64asm.BL || inst.Inst.Op == arm64asm.BLR
+}
+
+// IsRet returns true if the instruction is a RET or LRET instruction.
+func (inst *AsmInstruction) IsRet() bool {
+	if inst.Inst == nil {
+		return false
+	}
+	return inst.Inst.Op == arm64asm.RET || inst.Inst.Op == arm64asm.ERET
+}
+
+func resolveCallArg(inst *archInst, currentGoroutine bool, regs Registers, mem MemoryReadWriter, bininfo *BinaryInfo) *Location {
+	if inst.Op != arm64asm.BL /*&& inst.Op != arm64asm.B*/ {
+	//if inst.Op != arm64asm.ADRP && inst.Op != arm64asm.LDUR {
+		return nil
+	}
+
+	var pc uint64
+	var err error
+	switch arg := inst.Args[0].(type) {
+	case arm64asm.ArchArm64:
+		pc = uint64(arg)
+	case arm64asm.Reg:
+		if !currentGoroutine || regs == nil {
+			return nil
+		}
+		pc, err = regs.Get(int(arg))
+		if err != nil {
+			return nil
+		}
+
+	case arm64asm.RegSP:
+		fmt.Println("1")
+
+	case arm64asm.ImmShift:
+		fmt.Println("2")
+
+	case arm64asm.PCRel:
+		//pcbytes := make([]byte, inst.MemBytes)
+		//pc = binary.LittleEndian.Uint64(pcbytes)
+		fmt.Println("3")
+
+	case arm64asm.MemImmediate:
+		fmt.Println("4")
+
+	case arm64asm.MemExtend:
+		fmt.Println("5")
+
+	case arm64asm.Imm:
+		fmt.Println("6")
+
+	case arm64asm.Imm_hint:
+		fmt.Println("7")
+
+	case arm64asm.Imm_clrex:
+		fmt.Println("8")
+
+	case arm64asm.Imm_dcps:
+		fmt.Println("9")
+
+	case arm64asm.Cond:
+		fmt.Println("11")
+
+		// Reg, RegSP, ImmShift, RegExtshiftAmount, PCRel, MemImmediate,
+		// MemExtend, Imm, Imm64, Imm_hint, Imm_clrex, Imm_dcps, Cond,
+		// Imm_c, Imm_option, Imm_prfop, Pstatefield, Systemreg, Imm_fp
+		// RegisterWithArrangement, RegisterWithArrangementAndIndex.
+
+	case arm64asm.Imm_c:
+		fmt.Println("12")
+
+	case arm64asm.Imm_option:
+		fmt.Println("13")
+
+	case arm64asm.Imm_prfop:
+		fmt.Println("14")
+
+	case arm64asm.Pstatefield:
+		fmt.Println("15")
+
+	case arm64asm.Systemreg:
+		fmt.Println("16")
+
+	case arm64asm.Imm_fp:
+		fmt.Println("17")
+
+	case arm64asm.RegisterWithArrangement:
+		fmt.Println("18")
+
+	case arm64asm.RegisterWithArrangementAndIndex:
+		fmt.Println("19")
+
+	default:
+		return nil
+	}
+	file, line, fn := bininfo.PCToLine(pc)
+	if fn == nil {
+		return &Location{PC: pc}
+	}
+	return &Location{PC: pc, File: file, Line: line, Fn: fn}
+
+}
+
+type instrseq []arm64asm.Op
+
+// Possible stacksplit prologues are inserted by stacksplit in
+// $GOROOT/src/cmd/internal/obj/x86/obj6.go.
```

Comment:
This comment is wrong, clearly you didn't get this prologues from the x86 file. Also I did check the correct linker file and the stack split code looks different from what you checking here. If you don't feel like figuring out what the prologues actually look like remove all this code and reject programs built before 1.12 on arm64.